core/http-auth-aws/src/main/java/software/amazon/awssdk/http/auth/aws/internal/AwsS3V4HttpProperties.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.http.auth.aws.internal;
+
+import static software.amazon.awssdk.http.auth.AwsV4HttpSigner.CHUNKED_ENCODING;
+import static software.amazon.awssdk.http.auth.AwsV4HttpSigner.PAYLOAD_SIGNING;
+import static software.amazon.awssdk.http.auth.internal.util.SignerUtils.safeParseLong;
+import static software.amazon.awssdk.http.auth.internal.util.SignerUtils.validatedProperty;
+
+import java.time.Clock;
+import java.util.Optional;
+import software.amazon.awssdk.annotations.SdkProtectedApi;
+import software.amazon.awssdk.http.Header;
+import software.amazon.awssdk.http.auth.aws.AwsS3V4HttpSigner;
+import software.amazon.awssdk.http.auth.internal.AwsV4HttpProperties;
+import software.amazon.awssdk.http.auth.internal.checksums.ChecksumAlgorithm;
+import software.amazon.awssdk.http.auth.internal.util.CredentialScope;
+import software.amazon.awssdk.http.auth.internal.util.HttpChecksumUtils;
+import software.amazon.awssdk.http.auth.spi.SignRequest;
+import software.amazon.awssdk.identity.spi.AwsCredentialsIdentity;
+
+/**
+ * An extension of {@link AwsV4HttpProperties}, which provides access to more specific parameters
+ * used by {@link AwsS3V4HttpSigner}.
```

Comment:
What do we use it for?

Reply:
Everything AWS related is going to move to http-auth-aws, so there won't be cross-module.